service/dap/types.go

```diff
@@ -6,39 +6,68 @@ import (
 	"fmt"
 )
 
+func isValidLaunchMode(mode string) bool {
+	switch mode {
+	case "exec", "debug", "test":
+		return true
+	}
+	return false
+}
+
+// Default values for Launch/Attach configs.
+// Used to initialize configuration variables before decoding
+// arguments in launch/attach requests.
+var (
+	defaultLaunchAttachCommonConfig = LaunchAttachCommonConfig{
+		Backend:         "default",
+		StackTraceDepth: 50,
+	}
+	defaultLaunchConfig = LaunchConfig{
+		Mode:                     "debug",
+		Output:                   defaultDebugBinary,
+		LaunchAttachCommonConfig: defaultLaunchAttachCommonConfig,
+	}
+	defaultAttachConfig = AttachConfig{
+		Mode:                     "local",
+		LaunchAttachCommonConfig: defaultLaunchAttachCommonConfig,
+	}
+)
+
 // LaunchConfig is the collection of launch request attributes recognized by delve DAP implementation.
 type LaunchConfig struct {
+	// Acceptable values are:
+	//   "debug": compiles your program with optimizations disabled, starts and attaches to it.
+	//   "test": compiles your unit test program with optizations disabled, starts and attaches to it.
+	//   "exec": executes a precompiled binary and begin a debug session.
+	//
+	// Default is "debug".
+	Mode string `json:"mode"`
+
+	// Required when mode is `debug`, `test`, or `exec`.
+	// Path to the program folder (or any go file within that folder)
+	// when in `debug` or `test` mode, and to the pre-built binary file
+	// to debug in `exec` mode.
+	// If it is not an absolute path, it will be interpreted as a path
+	// relative to the working directory of the delve process.
+	Program string `json:"program,omitempty"`
+
 	// Command line arguments passed to the debugged program.
 	Args []string `json:"args,omitempty"`
 
-	// Backend used by delve. Maps to `dlv`'s `--backend` flag.
-	// Allowed Values: `"default"`, `"native"`, `"lldb"`
-	Backend string `json:"backend,omitempty"`
-
-	// Build flags, to be passed to the Go compiler. Maps to dlv's `--build-flags` flag.
-	BuildFlags string `json:"buildFlags,omitempty"`
-
 	// Absolute path to the working directory of the program being debugged
 	// if a non-empty value is specified. If not specified or empty,
 	// the working directory of the delve process will be used.
+	// This is similar to delve's `--wd` flag.
 	Cwd string `json:"cwd,omitempty"`
 
-	// Environment variables passed to the program.
-	Env map[string]string `json:"env,omitempty"`
-
-	// One of `debug`, `test`, `exec`.
-	Mode string `json:"mode,omitempty"`
+	// Build flags, to be passed to the Go compiler.
+	// For example, "-tags=integration -mod=vendor -cover -v".
```

Comment:
Like --build-flags

Reply:
Done.

Side note: I really wish we could accept []string  and skip delve's custom build flag parsing if possible. There are subtle differences between dlv and go's flag parsing.